### PR TITLE
Applied Routing & Redux in Todo List

### PR DESCRIPTION
**Pull request by ghulamhafeez**
## Routing

- Applied Routing in Todo list
### Routing Gif

- 
r

## Redux Toolkit

- Applied Redux Toolkit in Todo list & also create redux counter example

### CODE REVIEW

File: src/container/Contents.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    {
+        path:'/counter',
+        element:<Counter />
+    }
+  ]);
+
+const Contents = () => {
+  return (
+    <div>
```
Comment (Monfernape) [inline src/container/Contents.js]: Extra div.

File: src/reducers/CounterReducer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit'
+const initialStateValue={
+  count: 0
+}
+export const counterReducer = createSlice({
```
Comment (Monfernape) [inline src/reducers/CounterReducer.js:5]: Nice stuff

File: src/reducers/TodoReducer.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        title: action.payload,
+        id: Date.now(),
+        isCompleted: false,
+      };
+
+      state.todos = [...state.todos, myData];
+    },
+
+    onUpdate: (state, action) => {
+      let updatedTodos = state.todos.map((x) =>
```
Comment (Monfernape) [inline src/reducers/TodoReducer.js:24]: Great job.

File: src/reducers/TodoReducer.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        x.id === action.payload.id
+          ? { ...x, title: action.payload.todovalue }
+          : x
+      );
+
+      state.todos = updatedTodos;
+      action.payload.todovalue = "";
```
Comment (Monfernape) [inline src/reducers/TodoReducer.js]: Not needed

File: src/reducers/TodoReducer.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    onToggleCompletion: (state, action) => {
+      let checkIsCompleted = state.todos.map((x) =>
+        x.id === action.payload ? { ...x, isCompleted: !x.isCompleted } : x
+      );
+
+      state.todos = checkIsCompleted;
+    },
+    onGetTodos: (state, action) => {
+      state.todos = action.payload;
```
Comment (Monfernape) [inline src/reducers/TodoReducer.js:44]: Super cool

**Review by Monfernape — APPROVED** [review]
Looks good. Notice how much we've reduced our code

File: src/reducers/TodoReducer.js (hunk 5 above)
Comment (ghulamhafeez) [reply src/reducers/TodoReducer.js:44]: Thanks

File: src/reducers/TodoReducer.js (hunk 4 above)
Comment (ghulamhafeez) [reply src/reducers/TodoReducer.js]: ok

File: src/reducers/CounterReducer.js (hunk 2 above)
Comment (ghulamhafeez) [reply src/reducers/CounterReducer.js:5]: Thanks

File: src/reducers/TodoReducer.js (hunk 3 above)
Comment (ghulamhafeez) [reply src/reducers/TodoReducer.js:24]: Thanks

File: src/container/Contents.js (hunk 1 above)
Comment (ghulamhafeez) [reply src/container/Contents.js]: ok I will remove

**Pull request merged**